### PR TITLE
added: service utilities for history api call

### PR DESCRIPTION
**Pull request by rkapoor10**
## Serivce utilities added 

- postUserHistory
- getUserHistory
- history remains after reload
- updated local storage for liked and watchlater videos

### CODE REVIEW

File: src/backend/utils/authUtils.js [hunk 1]
```diff
@@ -4,10 +4,12 @@ import jwt_decode from "jwt-decode";
 
 export const requiresAuth = function (request) {
   const encodedToken = request.requestHeaders.authorization;
+  console.log(encodedToken)
   const decodedToken = jwt_decode(
     encodedToken,
     process.env.REACT_APP_JWT_SECRET
   );
+  console.log(decodedToken)
```
Comment (Kothalkarpriya) [inline src/backend/utils/authUtils.js:12]: If you have checked the token you can remove the console.log

**Review by Kothalkarpriya — APPROVED** [review]
Good work! The code is understandable. You can merge the changes

File: src/components/auth/LoginForm.jsx [hunk 2]
```diff
@@ -22,7 +22,6 @@ const LoginForm = () => {
     checkPolicy: true,
   };
   const loginBtnHandler = (e, user) => {
-    console.log(location, "location-login")
     e.preventDefault();
```
Comment (t007rushi) [inline src/components/auth/LoginForm.jsx:25]: if you do e.preventDef in call itslef no need to pass e here

File: src/components/auth/LoginForm.jsx [hunk 3]
```diff
@@ -129,7 +131,6 @@ const LoginForm = () => {
                     className="btn-secondary txt-s w-80"
                     onClick={(e) => {
                       e.preventDefault();
```
Comment (t007rushi) [inline src/components/auth/LoginForm.jsx:133]: ok you did it here so delete above as double prevention to same event not needed

File: src/services/history/postUserHistory.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      "/api/user/history",
+      {
+        video,
+      },
+      axiosConfig
+    );
+    return responseFromHistory;
+  } catch (error) {
+    console.error(error);
+    toast.error(error);
```
Comment (t007rushi) [inline src/services/history/postUserHistory.js:18]: need of both?

**Review by t007rushi — APPROVED** [review]
there are consoles if not needed or if future ref then comment so reviewer can found easy to point out or not
else can go ahead

File: src/backend/utils/authUtils.js (hunk 1 above)
Comment (rkapoor10) [reply src/backend/utils/authUtils.js:12]: will do

File: src/components/auth/LoginForm.jsx (hunk 2 above)
Comment (rkapoor10) [reply src/components/auth/LoginForm.jsx:25]: noted

File: src/components/auth/LoginForm.jsx (hunk 3 above)
Comment (rkapoor10) [reply src/components/auth/LoginForm.jsx:133]: will do

File: src/services/history/postUserHistory.js (hunk 4 above)
Comment (rkapoor10) [reply src/services/history/postUserHistory.js:18]: no 😂

**Pull request merged**